mirdata/irmas.py

```diff
@@ -0,0 +1,406 @@
+                                        ]  # Obtain id without '.wav'
+                                        split_dru = file.split('[')[3]
+                                        genre_code = split_dru.split(']')[0]
+                                        irmas_dict[irmas_id_dru_no_wav] = {
+                                            'genre': genre_code,
+                                            'drum': True,
+                                            'train': True,
+                                        }
+                                    if 'nod' in file:
+                                        irmas_id_nod = file.split(']')[3]  # Obtain id
+                                        irmas_id_nod_no_wav = irmas_id_nod.split('.')[
+                                            0
+                                        ]  # Obtain id without '.wav'
+                                        split_nod = file.split('[')[3]
+                                        genre_code = split_nod.split(']')[0]
+                                        irmas_dict[irmas_id_nod_no_wav] = {
+                                            'genre': genre_code,
+                                            'drum': False,
+                                            'train': True,
+                                        }
+                                    else:
+                                        irmas_id = file.split(']')[2]  # Obtain id
+                                        irmas_id_nod_no_wav = irmas_id.split('.')[
+                                            0
+                                        ]  # Obtain id without '.wav'
+                                        split_1 = file.split('[')[2]
+                                        genre_code = split_1.split(']')[0]
+                                        irmas_dict[irmas_id_nod_no_wav] = {
+                                            'genre': genre_code,
+                                            'drum': None,
+                                            'train': True,
+                                        }
+            if 'Test' in directory:
+                for root_, dirs_, files_ in os.walk(os.path.join(data_home, directory)):
+                    for directory_ in dirs_:
+                        for root__, dirs__, files__ in os.walk(
+                            os.path.join(data_home, directory, directory_)
+                        ):
+                            for file in files__:
+                                if file.endswith('.wav'):
+                                    test_index = str(count)
+                                    irmas_dict[test_index] = {
+                                        'genre': None,
+                                        'drum': None,
+                                        'train': False,
+                                    }
+                                    count += 1
+
+    irmas_dict["data_home"] = data_home
```

Comment:
I think this code can be simplified but I want to make sure I understand what you're doing first. I guess the metadata is contained in the `wavfile` name, right? So you use split to get the different components?

If I got that right, I think you can look over the track ids, and simply change the dict as you go:

``` 
for track_id in INDEX.keys():
        irmas_dict[track_id]={
                              'genre': split(INDEX[track_id].audio_path, bla bla). ,
                              'drum': [True if 'dru' in INDEX[track_id].audio_path else False][0],
                              'train': [True if 'Train' in INDEX[track_id].audio_path else False][0]
                              }
```
Or something like that. Does this make sense?

Reply:
I also think you should look at the index rather than at the files that are on the hard disk

Reply:
Yes, currently fixing it. I haven't though that I could use the json index here! However, I'll need to add some conditions as the files have to be split from different points because the 'dru' (has drum) or 'nod' (has not drum) annotation is not in every file.

Reply:
Solved! See [commit 45a71fd](https://github.com/mir-dataset-loaders/mirdata/pull/288/commits/45a71fd477006900ac2701e59c139968d3ab0e82).